nodejs/week3/app.js

```diff
@@ -0,0 +1,50 @@
+require('dotenv').config();
+
+const knex = require('knex')({
+  client: 'mysql2',
+  connection: {
+    host: process.env.DB_HOST || '127.0.0.1',
+    port: process.env.DB_PORT || 3306,
+    user: process.env.DB_USER || 'root',
+    password: process.env.DB_PASSWORD || 'my-secret-pw',
+    database: process.env.DB_NAME || 'hyf_node_week3_warmup',
+    multipleStatements: true,
+  },
+});
+
+const express = require('express');
+const app = express();
+const port = process.env.PORT || 3000;
+
+app.use(express.json());
+
+const apiRouter = express.Router();
+app.use('/api', apiRouter);
+
+const contactsAPIRouter = express.Router();
+apiRouter.use('/contacts', contactsAPIRouter);
+
+contactsAPIRouter.get('/', async (req, res) => {
+  let query = knex.select('*').from('contacts');
+
+  if ('sort' in req.query) {
+    const orderBy = req.query.sort.toString();
+    if (orderBy.length > 0) {
+      query = query.orderBy(orderBy);
```

Comment:
This does prevent SQL injections but is it possible to set the sort direction now? Does `GET /api/contacts?sort=last_name%20DESC` work?